server/server.js

```diff
@@ -153,7 +153,14 @@ app.post(`${apiBaseUrl}/approve-user`,
   keycloak.allowRolesMiddleware('ministry_of_health'),
   asyncMiddleware(async (req, res) => {
     await validate(AccessRequestApproval, req.body);
-    await keycloak.approvePendingRequest(req.body.userId, req.body.role, [req.body.region]);
+    const promises = req.body.sites.map((siteId) => dbClient.db[collections.EMPLOYER_SITES].findDoc({ siteId }));
+    const results = await Promise.all(promises);
+    const siteRegions = results.map(item => item[0].healthAuthority);
```

Comment:
Could avoid using the misleading array index `[0]` by using the `single` option of massive `findDoc(query, { single: true }`.